CONTRIBUTING_TO_OPENSHIFT_PLUGINS.md

```diff
@@ -0,0 +1,280 @@
+Set up your git remotes so origin is the jenkinsci repository, and upstream is the openshift repository.  Using openshift-pipeline as an example (and substitute the 
+other plugin names if working with those plugins):
+
+1. From the parent directory you've chosen for you local repository, run `git clone git@github.com:jenkinsci/openshift-pipeline-plugin.git`
+1. Change directories into `openshift-pipeline-plugin`, and run `git remote add upstream git://github.com/openshift/jenkins-plugin`
+1. Then pull and rebase the latest changes from [https://github.com/openshift/jenkins-plugin](https://github.com/openshift/jenkins-plugin) with the following:
+
+```
+	$ git checkout master	
+	$ git fetch upstream	
+	$ git fetch upstream --tags	
+	$ git rebase upstream/master	
+	$ git push origin master	
+	$ git push origin --tags
+```
+
+#### Commits between openshift and jenkinsci repositories are no longer in sync
+
+This condition now exists for the openshift-sync and openshift-client plugins.  As such, we have to `git cherry-pick`.  The recipe is same as above, except the 
+`git rebase upstream/master` is replaced with either a `git cherry-pick <commit id>` or `git cherry-pick -m 1 <merge commit id>` based on whether you are pulling 
+in a direct commit or merge commit.
+
+### Why the commits may be out of sync / Why the separate repositories / How to interpret the differences
+
+#### Testing:
+
+* We can employ our CI server and extended test framework for respositories in https://github.com/openshift.  We cannot for https://github.com/jenkinsci.
+
+#### Release cycle (when we get to the end of a release):
+
+* We can continue development of new features for the next release in the https://github.com/openshift repositories.  But if a new bug for the current release
+pops up in the interim, if need be we can make the change in https://github.com/jenkinsci first, then cherry-pick into https://github.com/openshift (employing
+our regression testing), and then when ready cut the new version of the plugin in https://github.com/jenkinsci.
+
+#### What version is a change really in:
+
+* Through various scenarios and human error, those "release commits" (where the version in the pom.xml is updated) sometimes have landed in https://github.com/openshift after the fact, or out of order with
+how they landed in https://github.com/jenkinsci.  Jenkinci is the *master* in this case.  The version a change is in is expressed by the commit orders in the jenkinsci release.
+
+### Submit the new release to the Jenkins organization
+
+Again: your Git ID has push access to the two repositories for this plugin; your Jenkins ID (https://wiki.jenkins-ci.org/display/JENKINS/User+Account+on+Jenkins) is listed in the permission file for the plugin, like https://github.com/jenkins-infra/repository-permissions-updater/blob/master/permissions/plugin-openshift-pipeline.yml.  Given these assumptions:
```

Comment:
s/Again/Prereq/
s/two repositories/two repositories(openshift org and jenkinsci org)/